src/main/java/org/spongepowered/api/entity/living/player/server/ServerPlayer.java

```diff
@@ -239,20 +239,29 @@ default boolean isViewingInventory() {
 
     /**
      * Gets the {@link WorldBorder} for this player, if present. If no border is
-     * set, {@link Optional#empty()} is returned.
+     * set, an empty {@code Optional} is returned.
      *
      * @return The {@code WorldBorder} of this player as an {@code Optional}, if
      *     present
      */
     Optional<WorldBorder> worldBorder();
 
     /**
-     * Sets the {@link WorldBorder} instance for this player to the given world
-     * border. If {@code null} is passed, the world border is unset.
+     * Sets the {@link WorldBorder} that this player sees. If {@code null}, the
+     * border is unset, reverting to the border of the world the player is
+     * currently in.
+     *
+     * <p>A {@code false} return does not mean a border was not set, rather,
+     * that the resultant border that was set is not the one that was supplied.
+     * If false is returned, it is suggested that you inspect
+     * {@link #worldBorder()} to investigate what was changed.
+     * </p>
```

Comment:
This seems to be a case of we should return the Optional<WorldBorder> object that was set (if it wasn't refused)

Reply:
Like, asking the developer to re-inspect after attempting to set seems to me that we should just provide it back a a result (Transaction if you will).

Reply:
I've just made it an `Optional<WorldBorder>`. If it fails, it will return the current border. If it wasn't changed, the supplied object and the returned object (in the optional) will be the same reference. I'll add that to the docs.